File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        vector<int> res(2,0);
        int sumA=0;
        int n=nums.size();
        unordered_map<int,int> m;
        for(int i=0;i<n;++i){
            m[nums[i]]++;
            sumA+=nums[i];
        }
        int sum=(n*(n+1))/2;
        //cout<<sum;
        for(int i=0;i<n;++i)
        {
            if(m[nums[i]]>1)
                res[0]=nums[i],res[1]=sum-(sumA-nums[i]);
        }
         
     return res;
    }
};

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int correctSum = (nums.length*(2 + nums.length-1))/2;
        int actualSum = 0;
        int badNum = 0;        
        for (int n : nums) {
            nums[Math.abs(n)-1] *= -1;
            if (nums[Math.abs(n)-1] > 0) {
                badNum = Math.abs(n);
            }
            actualSum += Math.abs(n);
        }
        return new int[]{badNum, correctSum - (actualSum - badNum)};
    }
}